includes/wc-payment-api/class-wc-payments-api-client.php

```diff
@@ -435,19 +419,6 @@ public function create_and_confirm_setup_intent( $payment_method_id, $customer_i
 			'confirm'        => 'true',
 		];
 
-		if ( WC_Payments_Features::is_sepa_enabled() ) {
-			$request['payment_method_types'] = [ Payment_Method::CARD, Payment_Method::SEPA ];
```

Comment:
I wonder if we need to add some logic here to add `sepa_debit` if the payment method type is SEPA. I tried to complete a checkout for a subscription product with a free-trial, to trigger this function, and chose a saved SEPA Debit payment method and the checkout succeeded, but the payment method displayed on the order complete page said "Credit / Debit Card". Also, when I checked the Order itself, there was no intent mentioned nor a link to the Stripe dashboard to see the intent details; the order page for the subscription itself was also a little broken, so I'm pretty sure something went wrong here, even though it completed successfully.

Reply:
I think you're right: tested again and pretty sure this flow doesn't touch `create_and_confirm_intention`, so I don't think we need to update anything here after all. 

However, I'm pretty sure something is still broken in this scenario, but it doesn't appear to be exclusive to SEPA: I think I can repeat this oddity using a saved payment method of any payment method type. I'm going to open a new issue to take care of this one...

Reply:
Created #3396.